src/services/db/GitFileSystemService.ts

```diff
@@ -912,7 +912,8 @@ export default class GitFileSystemService {
   listDirectoryContents(
     repoName: string,
     directoryPath: string,
-    branchName: string
+    branchName: string,
+    includeSha = true
```

Comment:
may i get the context for why we are adding this param?

Reply:
this param is required to skip/include the sha; if we skip teh sha, then we can delete safely without encountering a `409` error

Reply:
ZJ made a change to fast load images as huge repos with a lot of images had very slow load times. 

We used to do: get all files -> calculate hash -> filter/paginate
Now, we get all files -> filter/paginate -> calculate hash

For the get all files to be fast, we removed the sha calculation in the first step of get all files. However, this affected other ops which use list directory contents

Specifically, when we try to rename a subfolder, we need to delete the current directory and re-create a new one
When deleting the current directory, it uses list directory method, but the sha is not present. So when it does a verification of oldSha === newSha, it fails as oldSha is undefined. This causes BE to think that the file was recently changed, so it blocks the delete operation.

Hence, the rename op fails, and we do a rollback. We return 409 saying someone edited the file.

Reply:
ok understand this is to fix prod, so dont have to fix in this pr but i do have concerns of this API that it is confusing. would rather have some "force=true" than have `includeSha = true` but am not sure how that

Reply:
knowing the sha will allow oldSha === newSha be true for .keep and all files under the subdirectory, which allows the delete op to go through

Reply:
includeSha is more specific than "force=true". Includesha flag tells the method to compute sha for all the files it is fetching in listDirectoryContents